docs/recipes/real_time_proc.qmd

```diff
@@ -0,0 +1,107 @@
+---
+title: "Real-Time Processing"
+execute:
+  eval: false
+---
+
+
+This recipe serves as an example to showcase the real-time processing capability of DASCore. Here, we demonstrate how to use DASCore to perform rolling mean processing on a spool in real time for edge computing purposes.
+
+
+## Load libraries and get a spool
+
+```{python}
+# Import libraries
+import os
+import time
+
+import dascore as dc
+import numpy as np
+
+from dascore.units import s 
+
+
+# Define path for saving results
+output_data_folder = '/path/to/desired/output/directory'
+
+# Get a spool to work on
+sp = dc.get_example_spool().sort("time").update()
+```
+
+## Set real-time processing parameters (if needed)
+
+In this section, we define the window size and step size required for [rolling](https://dascore.org/api/dascore/proc/rolling/rolling.html) mean processing. With a sampling interval of 10 seconds, the cutoff frequency (Nyquist frequency) is determined to be 0.05 Hz. Additionally, we establish the desired wait time after each run by using the `sleep_time_mult` parameter, which acts as a multiplier coefficient for the number of seconds in each patch.
+
+```{python}
+# Define the target sampling interval (in sec.)
+dt = 10 
+
+# Determine window size 
+window = dt*s
+
+# Determine step size 
+step = dt*s
+
+# Set the desired wait time after each run
+sleep_time_mult = 1.2
+```
+
+## Real-time processing
+
+Eventually, we use a while loop to frequently call the spool and perform the processing. The while loop breaks if there are no new patches in the spool.
+
+```{python}
+# Start the for loop for real-time processing
+i = 0
+while True:
+    initial_run = (i == 0)
+    run_num = i+1
+    print(f"\nRun number: {run_num}")
+
+    # Select a updated sub-spool
+    sp = dc.spool(data_path).sort("time").update()
```

Comment:
I am not 100% sure, but  I don't think update will preserve the sorting, so I suggest you change to:

```python
sp = dc.spool(data_path).update().sort("time")
```

Also, any reason not to define the spool outside of the while loop so we don't have to init it each iteration?